src/expression.js

```diff
@@ -206,7 +206,7 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprAtom(refShorthandDefaultPos)
   if (refShorthandDefaultPos && refShorthandDefaultPos.start) return expr
-  return this.parseSubscripts(expr, startPos, startLoc)
+  return this.parseSubscripts(expr, startPos, startLoc, expr.type === "ArrowFunctionExpression" && this.parenthesesCount === 0)
```

Comment:
Yeah, that also doesn't help as arrow function starts with parentheses anyway :( Just trying to resist the idea that we need separate state and tracking in several places just for such a small addition, no luck so far...